src/temp_profile_collector.cpp

```diff
@@ -6,41 +6,69 @@ namespace duckdb {
 
 namespace {
 // Heuristic estimation of single IO request latency, out of which range are classified as outliers.
-constexpr double MIN_LATENCY_MILLISEC = 0;
-constexpr double MAX_LATENCY_MILLISEC = 1000;
-constexpr int LATENCY_NUM_BKT = 200;
+constexpr double MIN_READ_LATENCY_MILLISEC = 0;
+constexpr double MAX_READ_LATENCY_MILLISEC = 1000;
+constexpr int READ_LATENCY_NUM_BKT = 100;
+
+constexpr double MIN_OPEN_LATENCY_MILLISEC = 0;
+constexpr double MAX_OPEN_LATENCY_MILLISEC = 1000;
+constexpr int OPEN_LATENCY_NUM_BKT = 100;
+
+constexpr double MIN_GLOB_LATENCY_MILLISEC = 0;
+constexpr double MAX_GLOB_LATENCY_MILLISEC = 1000;
+constexpr int GLOB_LATENCY_NUM_BKT = 100;
 
 const string LATENCY_HISTOGRAM_ITEM = "latency";
 const string LATENCY_HISTOGRAM_UNIT = "millisec";
 } // namespace
 
-TempProfileCollector::TempProfileCollector() : histogram(MIN_LATENCY_MILLISEC, MAX_LATENCY_MILLISEC, LATENCY_NUM_BKT) {
-	histogram.SetStatsDistribution(LATENCY_HISTOGRAM_ITEM, LATENCY_HISTOGRAM_UNIT);
+TempProfileCollector::TempProfileCollector() {
+	histograms[static_cast<idx_t>(IoOperation::kRead)] =
```

Comment:
We could make a struct, which contains operation name, min value, max value and unit.